### PR TITLE
[MX-216] - Sign in with apple routes and callbacks, see details

### PR DESCRIPTION
**Pull request by brainbicycle**
https://artsyproduct.atlassian.net/browse/MX-216

- Fix spacing in account link error
- Add apple paths and Oauth calls
- Add passport-apple dependency
- Init strategy for apples sign in, add env vars
- Add routes for sign in with apple
- Add apple in example apple

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "ip": "^1.1.5",
     "mailcheck": "^1.1.1",
     "passport": "^0.3.0",
+    "passport-apple": "https://github.com/relivecc/passport-apple#cbe68f547c1e49fd07999a105651de690f18feec",
```
Comment (ashfurrow) [inline package.json]: Let's make sure to leave a comment on that pull request to indicate we found the change helpful 👍

File: lib/app/index.coffee [hunk 2]
```diff
@@ -42,6 +42,17 @@ module.exports = ->
     trackSignup('email'),
     ssoAndRedirectBack
 
+  # Apple OAuth
+  app.get opts.applePath, setCampaign, beforeSocialAuth('apple')
+  app.get opts.appleCallbackPath,
```
Comment (eessex) [inline lib/app/index.coffee]: Do we still need both `get` and `post` here?

Comment (brainbicycle) [reply lib/app/index.coffee]: Still looking into whether it is necessary, but I am thinking no.

Comment (brainbicycle) [reply lib/app/index.coffee]: Seems like a bug in the passport-apple library, both endpoints were being called when only post should have been. Using a different package now that seems to work as expected 👍

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@artsy/passport",
-  "version": "1.1.11",
-  "description": "Wires up the common auth handlers for Artsy's [Ezel](ezeljs.com)-based apps using [passport](http://passportjs.org/).",
+  "version": "1.1.15",
```
Comment (eessex) [inline package.json:3]: Why not 1.1.12?

Comment (brainbicycle) [reply package.json:3]: Good catch that should be 1.1.12, I'll update 👍

**Pull request merged**